sled-agent/src/sled_agent.rs

```diff
@@ -346,21 +346,33 @@ impl SledAgent {
             storage.zone_bundler().clone(),
         )?;
 
-        match config.vmm_reservoir_percentage {
-            Some(sz) if sz > 0 && sz < 100 => {
-                instances.set_reservoir_size(&hardware, sz).map_err(|e| {
-                    error!(log, "Failed to set VMM reservoir size: {e}");
-                    e
-                })?;
-            }
-            Some(sz) if sz == 0 => {
-                warn!(log, "Not using VMM reservoir (size 0 bytes requested)");
-            }
-            None => {
-                warn!(log, "Not using VMM reservoir");
+        // Configure the VMM reservoir as either a percentage of DRAM or as an
+        // exact size in MiB.
+        let reservoir_mode = match (
+            config.vmm_reservoir_percentage,
+            config.vmm_reservoir_size_mb,
+        ) {
+            (None, None) => ReservoirMode::None,
+            (Some(p), None) => ReservoirMode::Percentage(p),
+            (None, Some(mb)) => ReservoirMode::Size(mb),
+            (Some(_), Some(_)) => panic!(
+                "cannot specify vmm_reservoir_percentage and \
+                vmm_reservoir_size_mb at the same time"
+            ),
+        };
+
+        match reservoir_mode {
+            ReservoirMode::None => warn!(log, "Not using VMM reservoir"),
+            ReservoirMode::Size(0) | ReservoirMode::Percentage(0) => {
+                warn!(log, "Not using VMM reservoir (size 0 bytes requested)")
             }
-            Some(sz) => {
-                panic!("invalid requested VMM reservoir percentage: {}", sz);
```

Comment:
This check was here to verify that the agent would panic if it was configured with a percentage > 100. It looks like this got moved into an error into the caller, which I think is fine, but it would be good to put testing bad values in the config in some testing notes for the PR.

Reply:
I looked to see if there was a test that mocked some of this so that I could pass some values to the function but it doesn't look like such a setup exists.  Maybe we can use this as an excuse to jump on a bench gimlet and test this together?

Reply:
yeah, there's no unit test for that behavior. When I merged this PR initially, I just tested it ad hoc (since it's a pretty straightforward thing to test, and only matters for startup of the agent).

but yes, let's do that!